pkg/metrics/tool/stat.go

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2022. Nydus Developers. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package tool
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/containerd/containerd/log"
+)
+
+type Stat struct {
+	Utime  float64
+	Stime  float64
+	Cutime float64
+	Cstime float64
+	Thread float64
+	Start  float64
+	Rss    float64
+	Fds    float64
+	Uptime float64
+}
+
+const (
+	// constant value for linux platform except alpha and ia64
+	defaultClkTck = 100
+)
+
+var ClkTck = GetClkTck()
+var PageSize = float64(os.Getpagesize())
+
+func ParseFloat64(val string) float64 {
+	floatVal, err := strconv.ParseFloat(val, 64)
+	if err != nil {
+		log.L.Warnf("parse float failed, error: %v", err)
+	}
+	return floatVal
+}
+
+func GetClkTck() float64 {
+	getconfPath, err := exec.LookPath("getconf")
+	if err != nil {
+		log.L.Warnf("can not find getconf in PATH: %v", err)
+		return defaultClkTck
+	}
+	out, err := exec.Command(getconfPath, "CLK_TCK").Output()
+	if err != nil {
+		log.L.Warnf("get CLK_TCK failed: %v", err)
+		return defaultClkTck
+	}
+	return ParseFloat64(strings.ReplaceAll(string(out), "\n", ""))
+}
+
+func GetCurrentStat(pid int) *Stat {
+	uptimeBytes, err := os.ReadFile(path.Join("/proc", "uptime"))
+	if err != nil {
+		log.L.Warnf("get uptime failed: %v", err)
+		return nil
+	}
+	uptime := ParseFloat64(strings.Split(string(uptimeBytes), " ")[0])
+
+	statBytes, err := os.ReadFile(path.Join("/proc", strconv.Itoa(pid), "stat"))
+	if err != nil {
+		log.L.Warnf("get stat failed: %v", err)
+		return nil
+	}
+	splitAfterStat := strings.SplitAfter(string(statBytes), ")")
+
+	if len(splitAfterStat) == 0 || len(splitAfterStat) == 1 {
+		log.L.Warnf("can not find process, PID: %v", strconv.Itoa(pid))
+		return nil
+	}
+	infos := strings.Split(splitAfterStat[1], " ")
+
+	files, _ := os.ReadDir(path.Join("/proc", strconv.Itoa(pid), "fdinfo"))
```

Comment:
At least  give a explicit error message, don't ignore errors

Reply:
Sorry, fixed it.